### ISSUE TITLE
Evennia doesn't create telnet connections anymore

### ISSUE THREAD

**Issue by vlegoff** [evennia/evennia#1236]
#### Brief summary of issue / Description of requested feature:

`evennia start` doesn't create telnet connections anymore.  This occurs on a fresh install of Evennia with a fresh game directory and no changes in settings.  There is no logged error, either in the console or in a file.

#### Steps to reproduce the issue / Reasons for adding feature:

It is hard to say, because it seems to occur on my machine only and everything was working fine a few days ago.  But here's what I do:

1. Follow the documentation to install a fresh version of Evennia, including starting a new Python virtual-env.
2. Create a new, empty game directory with `evennia --init mygame`.
3. Start Evennia (with `evennia start`).  In the list of messages, there's no clear indication a connection is established on 4000, but no clear indication there's an error either.

#### Error output / Expected result of feature

There's unfortunately no clear error, either on the console or in any log files.  What I get on the console is the following:

```
Starting Evennia Server (output to stdout).

Starting Evennia Portal in Daemon mode (output to portal logfile).
--------------------------------------------------
 Even Portal (0.6.0 (rev effa618)) started.
  amp (to Server): 5000
  telnet: 4000
  webproxy:8000 (<-> 5001)
   + webclient:8001
--------------------------------------------------
--------------------------------------------------
 Mygame Server (0.6.0 (rev effa618)) started.
  amp (to Portal): 5000
  webserver: 5001
--------------------------------------------------
2017-02-25T14:22:22-0800 [-] Loading c:\users\vincent\evennia\evennia\server\server.py...
2017-02-25T14:22:25-0800 [-] Loaded.
2017-02-25T14:22:25-0800 [twisted.application.app.AppLogger#info] twistd 16.6.0 (c:\users\vincent\pyenv\scripts\python.exe 2.7.10) starting up.
2017-02-25T14:22:25-0800 [twisted.application.app.AppLogger#info] reactor class: twisted.internet.selectreactor.SelectReactor.
2017-02-25T14:22:25-0800 [-] AmpServerFactory starting on 5000
2017-02-25T14:22:25-0800 [-] Website starting on 5001
```

I've tried to use the port 4000 to bind a connection, everything seems to work.  Changing Evennia's telnet port doesn't seem to do anything (except alter one of the lines above).  Telnet indicates that the port is "closed" whenever I attempt to use it (the various MUD clients says the connection cannot be established).

When I run the webclient, it works until it tries to connect... then it displays unable to connect, like the other ones.  The website created on 5001 seems to work just fine.

#### Extra information, such as Evennia revision/repo/branch, operating system and ideas for how to solve / implement:

- I say it once more, but it worked perfectly well a few days ago, I don't know what changed on my system.
- Windows: 10 (64-bit)
- Evennia (lattest at this moment, commit effa6182683c81289b4822fcb0eee03f5d205b1a, although I tried a version with a commit 7 days ago and same result)

I don't know what's responsible for the error, but it's effective in preventing coding with Evennia at least.  Not sure if it's Evennia's fault, but I really don't know what's going on.  Ideas would be welcome!

**Comment by BlauFeuer** [comment]
In step 3 to reproduce, the clear indication is the third line:

>> Even Portal (0.6.0 (rev effa618)) started.
>>  amp (to Server): 5000
>>  **telnet: 4000**

indicating _listening_ on port 4000.  

Your complaint is, as far as I can tell, that a client is unable to _connect_ to port 4000.

It's true that the console will not show a connection, but it will show a login, so you can't get feedback from the console about telnet connections being established until it actually logs in.   Perhaps when debug is on, Evennia could display socket connections being made, which would help with diagnosis in your case.

Another thought is that since it last worked for you, perhaps a firewall rule was established, blocking incoming connections on that port or by the twistd process.

**Comment by vlegoff** [comment]
I think that when the server actually listens on 4000, after the line telling me the website is launched on port 5001, there's another telling me the telnet server is running on 127.0.0.1, port 4000.  Not anymore for me though.  In any ways, no connection (using telnet, a MUD client or the webclient) is possible.

I was looking for a debug flag to activate in the settings, but couldn't find it.  There's one (simply called `DEBUG`, that seems to be used only for the website).  I turned it on anyway.  The other error setting, to have the errors in the MUD client (which is no use here since there's not even a connection) is active by default.

I've disabled the Firewall to make sure.  I will try to turn off Windows Defender, although I don't see why it would prevent it from running.

**Comment by BlauFeuer** [comment]
Trying this myself. The full start output in console for me is:

>> 0 static files copied to '/home/evennia/mygame/web/static', 76 unmodified.
>> (pyenv) evennia@evenniatest:~/mygame$
>> Starting Evennia Server (output to stdout).
>> 
>> Starting Evennia Portal in Daemon mode (output to portal logfile).
>> --------------------------------------------------
>>  mygame Portal (0.6.0 (rev effa618)) started.
>>   amp (to Server): 5000
>>   telnet: 4000
>>   webproxy:8000 (<-> 5001)
>>    + webclient:8001
>> --------------------------------------------------
--------------------------------------------------
 >> mygame Server (0.6.0 (rev effa618)) started.
>>   amp (to Portal): 5000
>>   webserver: 5001
--------------------------------------------------
>> 2017-02-26T00:19:31+0001 [-] Loading /home/evennia/evennia/evennia/server/server.py...
>> 2017-02-26T00:19:32+0001 [-] Loaded.
>> 2017-02-26T00:19:32+0001 [twisted.scripts._twistd_unix.UnixAppLogger#info] twistd 16.2.0 (/home>> /evennia/pyenv/bin/python 2.7.11) starting up.
>> 2017-02-26T00:13:32+0001 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
>> 2017-02-26T00:19:32+0001 [-] AmpServerFactory starting on 5000
>> 2017-02-26T00:19:32+0001 [-] Website starting on 5001
>> 2017-02-26T00:19:33+0001 [evennia.server.amp.AmpServerFactory] AMPProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 5000) PEER:IPv4Address(TCP, '127.0.0.1', 33682))

Console only shows ports that are listening, and the amp connection connecting to 5000.  Telnet _is_ listening, but console output doesn't show anything until a successful login, not just a Telnet connection.  You could turn this into a feature request asking for telnet connections to be displayed when debug is on.

As-is, our log outputs show expected behavior and the console output doesn't help to show if any connections are successful to your telnet port.  Your connection attempts all seem to point to either the service is blocked or not reaching the correct host, and this can't be diagnosed from the console log.

**Comment by vlegoff** [comment]
I don't see your last line in my logs:

> 2017-02-26T00:19:33+0001 [evennia.server.amp.AmpServerFactory] AMPProtocol connection established (HOST:IPv4Address(TCP, '127.0.0.1', 5000) PEER:IPv4Address(TCP, '127.0.0.1', 33682))

Could it be the source of the problem?  That Evennia can't communicate between AMP and telnet for some reason?

**Comment by BlauFeuer** [comment]
It's a clue that perhaps localhost connections are failing for some reason.  It does seem suspicious!

**Comment by vlegoff** [comment]
That doesn't prove anything, but these connection settings seem to work though.

```
>>> import socket
>>> s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>>> s.bind(("127.0.0.1", 4000))
>>> s.listen(5)
>>> s2 = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>>> s2.bind(("localhost", 4000))
>>> s2.listen(5)
>>> s3 = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>>> s3.bind(("0.0.0.0", 4000))
>>> s3.listen(5)
>>>
```

**Comment by Griatch** [comment]
Just tried with a completely fresh clone and tried telnetting to localhost, port 4000 - works fine. You can also look at `server/logs/portal.log` - you should see one line saying `ServerFactory starting on 4000`. 

Either way, this seems to be a problem with your setup, not with Evennia itself, so I'm closing this from the issue list. You can keep replying here though if you want and we'll still see notifications in IRC (it will just not show in the main list).

**Comment by vlegoff** [comment]
Yeah, sounds like it's not an Evennia issue... but I have no idea how to 
fix it.  My Evennia is definitely broken on my system for some reason.  
I do have the message saying it starts listening on port 4000.  I have 
an error but that seems unrelated (and I don't know if I had it 
before).  Here it is:

```
2017-02-26T09:27:45-0800 [-] ServerFactory starting on 4000
2017-02-26T09:27:45-0800 [-] ServerFactory (WebSocketFactory) starting 
on 8001
2017-02-26T09:27:45-0800 [stderr#error] Traceback (most recent call last):
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\python27\Lib\runpy.py", line 174, in _run_module_as_main
2017-02-26T09:27:45-0800 [stderr#error]     "__main__", fname, loader, 
pkg_name)
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\python27\Lib\runpy.py", line 72, in _run_code
2017-02-26T09:27:45-0800 [stderr#error]     exec code in run_globals
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\scripts\twistd.exe\__main__.py", line 9, in <module>
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\scripts\twistd.py", 
line 29, in run
2017-02-26T09:27:45-0800 [stderr#error]     app.run(runApp, ServerOptions)
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\application\app.py", 
line 648, in run
2017-02-26T09:27:45-0800 [stderr#error]     runApp(config)
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\scripts\twistd.py", 
line 25, in runApp
2017-02-26T09:27:45-0800 [stderr#error] _SomeApplicationRunner(config).run()
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\application\app.py", 
line 383, in run
2017-02-26T09:27:45-0800 [stderr#error]     self.postApplication()
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\scripts\_twistw.py", 
line 48, in postApplication
2017-02-26T09:27:45-0800 [stderr#error] 
service.IService(self.application).privilegedStartService()
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\application\service.py", 
line 278, in privilegedStartService
2017-02-26T09:27:45-0800 [stderr#error] service.privilegedStartService()
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\application\internet.py", 
line 113, in privilegedStartService
2017-02-26T09:27:45-0800 [stderr#error]     self._port = self._getPort()
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\application\internet.py", 
line 141, in _getPort
2017-02-26T09:27:45-0800 [stderr#error]     'listen%s' % 
(self.method,))(*self.args, **self.kwargs)
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\internet\posixbase.py", 
line 478, in listenTCP
2017-02-26T09:27:45-0800 [stderr#error]     p.startListening()
2017-02-26T09:27:45-0800 [stderr#error]   File 
"c:\users\vincent\pyenv\Lib\site-packages\twisted\internet\tcp.py", line 
983, in startListening
2017-02-26T09:27:45-0800 [stderr#error]     raise 
CannotListenError(self.interface, self.port, le)
2017-02-26T09:27:45-0800 [stderr#error] 
twisted.internet.error.CannotListenError: Couldn't listen on 
0.0.0.0:8000: [Errno 10013] An attempt was made to access a socket in a 
way forbidden by its access permissions.
```

I don't know if it crashes the application or something.  Doesn't sound 
very likely, but perhaps, only perhaps, there's a link.

**Comment by vlegoff** [comment]
Okay, I tried to change the proxy port for the webserver... and, wonders 
of wonders, it works.  Something like:


     WEBSERVER_PORTS = [(12123, 5001)]

In the settings seem to fix the problem.  Thus I would suggest I opened 
a new issue to report that, should the proxy fail, Telnet doesn't start 
either, I don't know why it's related.  I'll have to find why my system 
doesn't want me to use port 8000, but perhaps it's a service, after all, 
this port is so often used.
